src/main/cpp/src/json_parser.cuh

```diff
@@ -610,85 +599,58 @@ class json_parser {
    *
    * @param str string to parse
    * @param to_match expected match str
-   * @param w_style the escape style for writing.
-   * @return a pair of success and length, where success is true if the string
-   * is valid and length is the number of bytes needed to encode the string
-   * in the given style.
+   * @return a tuple of values indicating if the parse process was successful, if field name was
+   *         matched, and a pointer to the past-end position of the parsed data
    */
-  static __device__ inline std::pair<bool, cudf::size_type> try_parse_string(
-    char_range_reader& str,
-    char_range_reader to_match = char_range_reader(char_range::null()),
-    escape_style w_style       = escape_style::UNESCAPED)
+  static __device__ inline thrust::tuple<bool, bool, char const*> try_parse_string(
+    char_range_reader str, char_range_reader to_match = char_range_reader(char_range::null()))
   {
-    if (str.eof()) { return std::make_pair(false, 0); }
-    char const quote_char = str.current_char();
-    int output_size_bytes = 0;
-
-    // write the first " if write style is escaped
-    if (escape_style::ESCAPED == w_style) { output_size_bytes++; }
+    if (str.is_empty()) { return thrust::make_tuple(false, false, nullptr); }
+    char const quote_char   = str.current_char();
+    bool matched_field_name = !to_match.is_null();
 
     // skip left quote char
     // We don't need to actually verify what it is, because we just read it.
     str.next();
 
     // scan string content
-    while (!str.eof()) {
+    while (!str.is_empty()) {
       char c = str.current_char();
       int v  = static_cast<int>(c);
-      if (c == quote_char) {
-        // path 1: match closing quote char
+      if (c == quote_char) {  // path 1: match closing quote char
         str.next();
-
-        // match check, the last char in match_str is quote_char
-        if (!to_match.is_null() && !to_match.eof()) { return std::make_pair(false, 0); }
-
-        // write the end " if write style is escaped
-        if (escape_style::ESCAPED == w_style) { output_size_bytes++; }
-
-        return std::make_pair(true, str.pos());
-      } else if (v >= 0 && v < 32) {
-        // path 2: unescaped control char
-
-        // copy if enabled, escape mode, write more chars
-        if (escape_style::ESCAPED == w_style) {
-          int escape_chars = escape_char(str.current_char(), nullptr);
-          output_size_bytes += (escape_chars - 1);
-        }
-
-        // check match if enabled
-        if (!try_match_char(to_match, str.current_char())) { return std::make_pair(false, 0); }
-
+        matched_field_name = matched_field_name && (to_match.is_null() || to_match.is_empty());
```

Comment:
This is an expression on boolean variable. In C++, such expressions can be short-circuit: the left operand is evaluated first, then the right operand is evaluated only if the left is `true`. So this means we can avoid the overhead of evaluating the right side (`to_match.is_null() ||...`) if `matched_field_name` is already `false`.

Because of such short-circuit functionality, there no exists the operators like `&&=`, `||=` etc.